src/Controls/src/Core/Handlers/Items2/iOS/ItemsSourceFactory2.cs

```diff
@@ -22,7 +22,7 @@ public static ILoopItemsViewSource CreateForCarouselView(IEnumerable itemsSource
 			IList _ when itemsSource is INotifyCollectionChanged => new LoopObservableItemsSource2(itemsSource as IList, collectionViewController, loop),
 			IEnumerable _ when itemsSource is INotifyCollectionChanged => new LoopObservableItemsSource2(itemsSource as IEnumerable, collectionViewController, loop),
 			IEnumerable<object> generic => new LoopListSource2(generic, loop),
-			_ => new LoopListSource(itemsSource, loop),
+			_ => new LoopListSource2(itemsSource, loop),
```

Comment:
Could you include a related test?

Reply:
Hi @jsuarezruiz, As of now, the test cases are running with CarouselView1 in CI. Once this (#[29315](https://github.com/dotnet/maui/pull/29315) ) PR got merged, I will replace the `CarouselView1` to `CarouselView2`